### PR TITLE
Ty use moment.js locale if already defined #2140

### PR DESCRIPTION
**Pull request by upmauro**
Feature #2140

### CODE REVIEW

File: components/calendar/calendar.ts [hunk 1]
```diff
@@ -221,13 +223,20 @@ export class Calendar implements AfterViewInit,OnInit,OnDestroy,ControlValueAcce
...
-    @Input() locale: LocaleSettings = {
+    @Input() locale: LocaleSettings = !moment ? {
         firstDayOfWeek: 0,
         dayNames: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"],
         dayNamesShort: ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
         dayNamesMin: ["Su","Mo","Tu","We","Th","Fr","Sa"],
         monthNames: [ "January","February","March","April","May","June","July","August","September","October","November","December" ],
         monthNamesShort: [ "Jan", "Feb", "Mar", "Apr", "May", "Jun","Jul", "Aug", "Sep", "Oct", "Nov", "Dec" ]
+    } : {
+        firstDayOfWeek: moment.localeData()._week.dow,
```
Comment (jessepinho) [inline components/calendar/calendar.ts]: This looks like a private property, however...?

File: components/calendar/calendar.ts [hunk 2]
```diff
@@ -221,13 +223,20 @@ export class Calendar implements AfterViewInit,OnInit,OnDestroy,ControlValueAcce
...
+        dayNames: moment.weekdays(),
+        dayNamesShort: moment.weekdaysShort(),
+        dayNamesMin: moment.weekdaysShort(),
+        monthNames: moment.months(),
+        monthNamesShort: moment.weekdaysShort()
```
Comment (jessepinho) [inline components/calendar/calendar.ts]: Should be `moment.monthsShort()`

File: components/calendar/calendar.ts [hunk 3]
```diff
@@ -221,13 +223,20 @@ export class Calendar implements AfterViewInit,OnInit,OnDestroy,ControlValueAcce
...
         dayNames: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"],
         dayNamesShort: ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
         dayNamesMin: ["Su","Mo","Tu","We","Th","Fr","Sa"],
         monthNames: [ "January","February","March","April","May","June","July","August","September","October","November","December" ],
         monthNamesShort: [ "Jan", "Feb", "Mar", "Apr", "May", "Jun","Jul", "Aug", "Sep", "Oct", "Nov", "Dec" ]
+    } : {
+        firstDayOfWeek: moment.localeData()._week.dow,
+        dayNames: moment.weekdays(),
+        dayNamesShort: moment.weekdaysShort(),
+        dayNamesMin: moment.weekdaysShort(),
```
Comment (jessepinho) [inline components/calendar/calendar.ts]: Should be `moment.weekdaysMin()`

File: components/calendar/calendar.ts (hunk 1 above)
Comment (jessepinho) [inline components/calendar/calendar.ts]: Should be `moment.localeDate().firstDayOfWeek()`.

File: components/calendar/calendar.ts [hunk 4]
```diff
@@ -221,13 +223,20 @@ export class Calendar implements AfterViewInit,OnInit,OnDestroy,ControlValueAcce
     
     @Output() onSelect: EventEmitter<any> = new EventEmitter();
     
-    @Input() locale: LocaleSettings = {
+    @Input() locale: LocaleSettings = !moment ? {
```
Comment (jessepinho) [inline components/calendar/calendar.ts]: This will cause a `ReferenceError`. Do something like `typeof moment === 'undefined'` instead.

**Comment by cagataycivici** [comment]
Rather not have a direct dependency to moment, thank you.

**Comment by upmauro** [comment]
@cagataycivici sorry, but maybe we can try use if moment is defined, i don't know if this is a good pratice, but this method will not create a "direct" dependency.

**Comment by jessepinho** [comment]
@upmauro This can also be done in your app (rather than in this library itself) by simply passing in this moment config object via the `locale` input. Does that make sense?

**Comment by bhaidar** [comment]
@upmauro Great idea. Can we find all the info needed by Calendar in the moment() object? Thanks

**Pull request closed without merging**